### DISCUSSION TITLE
Get Started: Translations 🌐

### DISCUSSION THREAD

**Discussion by Jeffser** [Jeffser/Alpaca#153] [General]
> [!NOTE]
> This discussion has been closed, if you want to contribute translations to Alpaca please read [this article](https://jeffser.com/alpaca/contributing-translations.html). Thanks for the interest!

**Comment by niekdomi** [comment]
Hey, I would like to to a translation into german :)

**Reply by Jeffser** [reply]
Hi, I will get a file for you ready!

Thanks

**Reply by Jeffser** [reply]
[Your file](https://github.com/Jeffser/Alpaca/blob/main/po/de.po) is ready!

These files consists of lines like this one

```
#: data/com.jeffser.Alpaca.desktop.in:3
#: data/com.jeffser.Alpaca.metainfo.xml.in:7
msgid "Alpaca"
msgstr ""
```
Your translations go inside of the quotes in `msgstr`

Some have multiple lines like this one

```
#: data/com.jeffser.Alpaca.metainfo.xml.in:71
msgid ""
"Removed DOCX compatibility temporally due to error with python-lxml "
"dependency"
msgstr ""
```


Once the file is pushed I will turn it on for the next version~!

thanks

**Reply by Jeffser** [reply]
if you have any questions feel free to ask on this thread!

**Reply by MehrzweckMandala** [reply]
> [Your file](https://github.com/Jeffser/Alpaca/blob/main/po/de.po) is ready!
> 
> These files consists of lines like this one
> 
> ```
> #: data/com.jeffser.Alpaca.desktop.in:3
> #: data/com.jeffser.Alpaca.metainfo.xml.in:7
> msgid "Alpaca"
> msgstr ""
> ```
> 
> Your translations go inside of the quotes in `msgstr`
> 
> Some have multiple lines like this one
> 
> ```
> #: data/com.jeffser.Alpaca.metainfo.xml.in:71
> msgid ""
> "Removed DOCX compatibility temporally due to error with python-lxml "
> "dependency"
> msgstr ""
> ```
> 
> Once the file is pushed I will turn it on for the next version~!
> 
> thanks

Since it seems that nothing more is coming - would it be possible to simply use the provided language file and continue?

**Reply by Jeffser** [reply]
Sure go ahead

**Comment by olumolu** [comment]
@Jeffser can you update hi.po.

**Reply by Jeffser** [reply]
alr

**Reply by Jeffser** [reply]
btw how's hi.po going? tell me when you want me to add it to the app!

**Reply by olumolu** [reply]
yes i have open a pull request i need some more time but still you can add hi.po i will push updates slowly

**Comment by loulou64490** [comment]
I have a small question, what is the meaning of the capital word in this sentence, bc I don't understand the sentence
> Some models require a modelfile, Alpaca fills FROM and SYSTEM (context) instructions automatically.

**Reply by Jeffser** [reply]
FROM and SYSTEM are variables that Ollama uses to define the model's behavior, just let them in english

**Reply by Jeffser** [reply]
Translate 'context' tho

**Reply by loulou64490** [reply]
okay bro

**Comment by rashadgasimli** [comment]
Hello, I would like to translate Azerbaijani language. Thanks.

**Reply by Jeffser** [reply]
Hi do you want to translate to Azerbaijani or just Azerbaijan Azerbaijani? I'm not familiar with the language and I'm not sure if it has regional differences

**Reply by rashadgasimli** [reply]
There are 3 alphabetical version of Azerbaijani language: Latin, Cyrilic and Perso-Arabic. I want to translate Azerbaijani (Latin).

**Reply by Jeffser** [reply]
Alright I prepared [this file](https://github.com/Jeffser/Alpaca/blob/main/po/az.po) for the translation
I need you to add your name and email (optional) in lines 4 and 12, this is just so that it keeps track that you are the one translating that language

Now, the syntax of the file, you told me you are not familiar with the way .po files work so let me give you an explanation.

Every string inside the project is represented by a `msgid` you will see a lot of strings in your file for example

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""
```

The translated string goes inside msgstr, for example in spanish that would be


```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"
```

Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

Now, you will probably notice that there's hundreds of strings, I don't expect you to manually translate each one so here's the deal.

From line 19 all the way to line 1119, it's just release notes and metadata for Flathub
And from line 1312 to 1876 it's just model descriptions.

You can translate them using Google Translate or an AI model.

Just make sure that everything makes sense. The rest of the file are actual strings that are used on the interface of Alpaca.

If you have any questions feel free to ask and I will answer them tomorrow (it's 2 am in my country so I'm going to sleep lol)

**Reply by Jeffser** [reply]
Also, if you have any questions about the syntax I recommend you to look at the [spanish translation](https://github.com/Jeffser/Alpaca/blob/main/po/es.po)

**Reply by rashadgasimli** [reply]
Thanks, I already translated ~1000 strings from "Alpaca.pot" file (including release notes and model descriptions)

**Comment by YusaBecerikli** [comment]
Hello, I want to translate Turkish language. Thanks.

**Reply by Jeffser** [reply]
Hi, thanks for your time, I prepared [this file](https://github.com/Jeffser/Alpaca/blob/main/po/tr.po)
I'll need you to fill lines 4 and 12, the email thing is optional, just fill it with your name

I will give you a quick explanation of po files

Every string inside the project is represented by a msgid you will see a lot of strings in your file for example

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""
```

The translated string goes inside msgstr, for example in spanish that would be

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"
```

Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

There are a lot of strings inside this file, it contains metadata for Flathub, release notes and of course the interface of the app itself, I don't expect you to translate all the strings manually, so here's the deal

From line 19 to 1119 it's just metadata and release notes and from line 1312 to 1876 it's just models descriptions, you can translate those lines using AI, Google Translate etc etc since they can be literally translated with no adaptation.

Just make sure everything makes sense and take all the time you need.

Also please be aware that I will be crediting you as 'YusaBecerikli' if you want me to use a different name please tell me before making the pull request of all the translations.

**Reply by loulou64490** [reply]
@YusaBecerikli For model description I recommend to use chatGPT
(if the generation stop, tell him to continue eg: continue the translation)
Eg:
```
Translate these descriptions of AI models into Turkish:

Llama 3.1 is a new state-of-the-art model from Meta available in 8B, 70B and 405B parameter sizes.
Google Gemma 2 is now available in 2 sizes, 9B and 27B.
[...]
MathΣtral: a 7B model designed for math reasoning and scientific discovery by Mistral AI.
An open weights function calling model based on Llama 3, competitive with GPT-4o function calling capabilities.
A 3.8B model fine-tuned on a private high-quality synthetic dataset for information extraction, based on Phi-3.
```

**Reply by loulou64490** [reply]
@YusaBecerikli And for translating I recommend using [Gtranslator](https://flathub.org/apps/org.gnome.Gtranslator)
Just download your .po file then it will automatically open with this app and you will have a nice interface and keyboard shortcuts. It's really easier

And when you are done, you can open your .po file with a simple text editor, copy all then paste directly on the GitHub editor and make your pull request

**Reply by Jeffser** [reply]
> @YusaBecerikli For model description I recommend to use chatGPT (if the generation stop, tell him to continue eg: continue the translation) Eg:

Aww you would use chatGPT instead of Alpaca? you meanie hahahaha

**Reply by loulou64490** [reply]
hahaha but gpt-4o is really overpowered

**Comment by OriginalSimon** [comment]
I am want to translate into Ukrainian.

**Reply by olumolu** [reply]
Is not russian and Ukrainian are same language.

**Reply by OriginalSimon** [reply]
No, these are two different languages. 
P.S
https://blog.duolingo.com/ukraine-language/

**Reply by olumolu** [reply]
Ohhh i see

**Reply by Jeffser** [reply]
Hi sorry for that comment from the other user, I will get an Ukrainian file ready as soon as possible

**Reply by Jeffser** [reply]
Alright, [this](https://github.com/Jeffser/Alpaca/blob/main/po/uk.po) is your file

I need you to add your name and email (optional) to lines 4 and 12.

Translations files include strings for the package metadata (all information shown in Flathub), AI model descriptions and of course the interface itself.

Most of the strings are just metadata and descriptions, these don't need any context to be translated so you can use a translator or AI for them

From line 19 to 1119 it's metadata
From line 1312 to 1876 it's model descriptions

The rest are interface strings so please translate them manually or just make sure they all make sense.

I've been told that some people use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to edit the file but I haven't test it myself. If you want to edit the file manually this is an example in spanish:

```
#: src/window.py:878
msgid "Remove '{}'"
msgstr "Remover '{}'"
```

Notice how I kept the '{}', this is used to replace text on a string, in that case that would be the name of a model, so in the interface it would be something like " Remover 'Alpaca' "

Once you make the pull request I will credit you on the repo and the app itself with your name on Github, if you want me to use a different name please tell me, if you have any questions feel free to tell me, thanks for your help!

**Reply by olumolu** [reply]
> Hi sorry for that comment from the other user, I will get an Ukrainian file ready as soon as possible

Yes i am also sorry. i searched yes though this two language do have similarities but not same.
Maybe some vocabs are similar. But language is different.

**Comment by oritwoen** [comment]
I'll gladly upload a translation for Polish ;)

**Reply by Jeffser** [reply]
Hi thanks for your interest, I've prepared [this file](https://github.com/Jeffser/Alpaca/blob/main/po/pl.po) for the polish translation.
I'll need you to fill lines 4 and 12, the email thing is optional, just fill it with your name

Some people like to use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to fill the file, in case you want to do it manually this is a round out of how it works

Every string inside the project is represented by a msgid you will see a lot of strings in your file for example

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""
```

The translated string goes inside msgstr, for example in spanish that would be

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"
```

Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

----

The file contains strings for the metadata used in storefronts (Flathub), the descriptions of the available models and of course the interface of Alpaca.

Now, there are a lot of lines dedicated to the descriptions and metadata, those lines don't require any context to be translated, in those cases you can use a translator or AI to make your life easier, these are the lines:

19 to 1303: Metadata and release notes
1368 to 2165: Model descriptions

Everything else is the interface itself.

Thanks for your offer to translate the app, if you have any questions feel free to ask them in this thread, once you are finished please make a pull request with the file.

Also, please be aware that I will credit you using your public name on GitHub on this discussion, the repository's README file and the app itself, if you want me to use a different name please tell me before making the pull request, thanks

**Reply by K0-RR** [reply]
@Jeffser I also can translate to the Polish language.
Do you require a 100% translation to be done before uploading or it can be done in parts (a few pull requests)?

**Reply by oritwoen** [reply]
No need @K0-RR , I still working on full translation, I should end tomorrow.

**Comment by spaciouskarter78** [comment]
I would like to provide localisation for Telugu (te)

**Reply by olumolu** [reply]
https://github.com/Jeffser/Alpaca/blob/main/po/alpaca.pot
Make a copy of that file and name it as te.po
Every string inside the project is represented by a msgid you will see a lot of strings in your file for example

#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""

The translated string goes inside msgstr, for example in spanish that would be

#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"

Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

The file contains strings for the metadata used in storefronts (Flathub), the descriptions of the available models and of course the interface of Alpaca.

Now, there are a lot of lines dedicated to the descriptions and metadata, those lines don't require any context to be translated, in those cases you can use a translator or AI to make your life easier, these are the lines:

19 to 1303: Metadata and release notes
1368 to 2165: Model descriptions

**Reply by spaciouskarter78** [reply]
> https://github.com/Jeffser/Alpaca/blob/main/po/alpaca.pot Make a copy of that file and name it as tg.po Every string inside the project is represented by a msgid you will see a lot of strings in your file for example
> 
> #: src/dialogs.py:75 msgid "Renaming '{}'" msgstr ""
> 
> The translated string goes inside msgstr, for example in spanish that would be
> 
> #: src/dialogs.py:75 msgid "Renaming '{}'" msgstr "Renombrando '{}'"
> 
> Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.
> 
> The file contains strings for the metadata used in storefronts (Flathub), the descriptions of the available models and of course the interface of Alpaca.
> 
> Now, there are a lot of lines dedicated to the descriptions and metadata, those lines don't require any context to be translated, in those cases you can use a translator or AI to make your life easier, these are the lines:
> 
> 19 to 1303: Metadata and release notes 1368 to 2165: Model descriptions

The internationally recognised and normally used language code for Telugu is te, i suggest you to change it from tg to te. I will begin my work once it's done

**Reply by olumolu** [reply]
That is totally ok use te.po

**Reply by spaciouskarter78** [reply]
do you need a fully translated file before committing or a partially translated file is fine too? 564 strings is a lot of time

**Reply by olumolu** [reply]
Take your time and push them it will be done.

> 19 to 1303: Metadata and release notes
1368 to 2165: Model descriptions

**Reply by spaciouskarter78** [reply]
> Take your time and push them it will be done.
> 
> > 19 to 1303: Metadata and release notes
> > 1368 to 2165: Model descriptions

#320 

I opened a pull request

**Comment by edo0** [comment]
Hi! I would like to provide localisation for Italian

**Reply by Jeffser** [reply]
Hi, thanks! I'll get a file ready for you

**Reply by Jeffser** [reply]
Alright I made [this file](https://github.com/Jeffser/Alpaca/blob/main/po/it.po)

I'll need you to fill lines 4 and 12, the email thing is optional, just fill it with your name

Some people like to use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to fill the file, in case you want to do it manually this is a round out of how it works

Every string inside the project is represented by a msgid you will see a lot of strings in your file for example

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""
```

The translated string goes inside msgstr, for example in spanish that would be

```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"
```

Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

----

The file contains strings for the metadata used in storefronts (Flathub), the descriptions of the available models and of course the interface of Alpaca.

Now, there are a lot of lines dedicated to the descriptions and metadata, those lines don't require any context to be translated, in those cases you can use a translator or AI to make your life easier, these are the lines:

19 to 1303: Metadata and release notes
1368 to 2165: Model descriptions

Everything else is the interface itself.

----

Thanks for your interest in translating the app, if you have any questions feel free to ask in this thread and I'll answer as soon as possible!

**Comment by avds2** [comment]
Hi! I'll be happy to translate this into Farsi. ^^

**Reply by rashadgasimli** [reply]
You mean Persian, right?

**Reply by avds2** [reply]
Yes. They are the same (:
fa_IR

**Comment by 666xan666** [comment]
Hi,
I can try translate into Polish [PL].

**Reply by olumolu** [reply]
Currently the main dev is on a sick leave i am creating the file that you can fork and push the translations.
Add your translation to msgstr from msgid

**Reply by 666xan666** [reply]
Got it. W8ing.

**Reply by olumolu** [reply]
You can start translating.
Every string inside the project is represented by a msgid you will see a lot of strings in your file for example
```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr ""
```
The translated string goes inside msgstr, for example in spanish that would be
```
#: src/dialogs.py:75
msgid "Renaming '{}'"
msgstr "Renombrando '{}'"
```
Please note that '{}' is kept in the string as that is used by the code to add something to the string for example in that case {} would be replaced by the name of the chat.

The file contains strings for the metadata used in storefronts (Flathub), the descriptions of the available models and of course the interface of Alpaca.

Now, there are a lot of lines dedicated to the descriptions and metadata, those lines don't require any context to be translated, in those cases you can use a translator or AI to make your life easier, these are the lines:

19 to 1303: Metadata and release notes
1368 to 2165: Model descriptions

Everything else is the interface itself.

Thanks for your interest in translating the app, if you have any questions feel free to ask in this thread and I'll answer as soon as possible!

**Reply by 666xan666** [reply]
@olumolu OK but where is the file? Which one should I edit? (YES, it is my 1st time)

**Reply by olumolu** [reply]
https://github.com/olumolu/Alpaca/blob/main/po/pl.po

**Comment by ovari** [comment]
Please add:
```
msgid "Hungarian (Hungary)"
msgstr "hu-HU magyar (Magyarország)"
```

Thank you

**Reply by olumolu** [reply]
?

**Comment by UnconnectedBedna** [comment]
Throw in an ~`se.po`~ `sv.po` file too and I'll fork, translate and send pr.
Swedish.. that is....

**Reply by olumolu** [reply]
Fork and create a se.po with alpaca.pot

**Comment by blackopsrepl** [comment]
Hey, I'll do italian if you want!

**Reply by edo0** [reply]
Hi! To avoid duplication, I have almost completed the Italian translation (20% left to go), was planning to upload it by the end of the weekend

**Reply by blackopsrepl** [reply]
Thanks for letting me know! Looking forward to see it!

**Reply by edo0** [reply]
@blackopsrepl Ciao!
I have sent through my commit [https://github.com/Jeffser/Alpaca/pull/376](https://github.com/Jeffser/Alpaca/pull/376) . 
Please feel free to bring up any suggestions or improvements you think may enhance the translation.

**Comment by Cyanide31k7** [comment]
I'm willing to translate to Vietnamese 🇻🇳!

**Reply by Jeffser** [reply]
Hi thanks, I prepared a [Vietnamese file](https://github.com/Jeffser/Alpaca/blob/main/po/vi.po) for you.

I'll need you to fill lines 4 and 12, the email thing is optional, just fill it with your name.

You can use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to help you manipulate the file.

If you want to manipulate the file directly this is a quick explanation of how it works:

Each translatable line on the Flathub Metadata and the app it self is represented like this:

```PO
#: src/window.ui:93
msgid "Loading Instance"
msgstr ""
```

There you can see where the line is used and what the line is, your translation must go inside the quotes in msgstr. So something like this:

```PO
#: src/window.ui:93
msgid "Loading Instance"
msgstr "Cargando Instancia"
```

There are some lines that are bigger, notice how you don't just move to the next line in the file but also open new quotes for it

```PO
#: src/window.ui:150
msgid ""
"Warning: Power saver mode is enabled, this will slow down message generation"
msgstr ""
"Advertencia: el modo de ahorro de energía está habilitado, esto ralentizará "
"la generación de mensajes."
```

And finally, there are some lines that have quotes inside of them, in those cases the quote is `\"` instead of just `"`

```PO
#: src/window.ui:331
msgid ""
"The default model to use on new chats and when Alpaca is launched with the "
"option --ask \"message\""
msgstr ""
"El modelo por defecto a usar en chats nuevos y cuando Alpaca es lanzado con "
"la opción --ask \"mensaje\""
```

-----

Now either way you want to do it, please make sure that the translations are good, you can use AI or a translator program for the patch notes and model descriptions but please make sure to write some good translations for the UI and Flathub metadata.

-----

There's no rush to get this done for a certain date or version, you can take your time and work on this when you can. Once you are finished just open a new pull request, I will check for anything out of place and then I will add it officially to Alpaca and credit you on the about dialog in the app itself and the repo's README.

Thanks again!

**Comment by sh1d0re** [comment]
Hello there, I can help you with the Japanese language if you liked. No worries, I am a native Japanese speaker, if you are concerned.

**Reply by sh1d0re** [reply]
Postscript:
There may be a problem with the Japanese language with some terminal fonts, which makes it look "weird". Same applies to CJK languages (Chinese (Mandarian&Cantonese) Japanese Korean).

**Reply by Jeffser** [reply]
I see, thanks for telling me, terminal fonts are only used if the bot outputs code, I think most of the time that would be in english so I don't think it would be a problem.

Thanks for wanting to translate Alpaca to Japanese!

I prepared [this file](https://github.com/Jeffser/Alpaca/blob/main/po/ja.po) for your translations.

I'll need you to fill lines 4 and 12, the email thing is optional, just fill it with your name.

You can use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to help you manipulate the file.

If you want to manipulate the file directly this is a quick explanation of how it works:

Each translatable line on the Flathub Metadata and the app it self is represented like this:

```PO
#: src/window.ui:93
msgid "Loading Instance"
msgstr ""
```

There you can see where the line is used and what the line is, your translation must go inside the quotes in msgstr. So something like this:

```PO
#: src/window.ui:93
msgid "Loading Instance"
msgstr "Cargando Instancia"
```

There are some lines that are bigger, notice how you don't just move to the next line in the file but also open new quotes for it

```PO
#: src/window.ui:150
msgid ""
"Warning: Power saver mode is enabled, this will slow down message generation"
msgstr ""
"Advertencia: el modo de ahorro de energía está habilitado, esto ralentizará "
"la generación de mensajes."
```

And finally, there are some lines that have quotes inside of them, in those cases the quote is `\"` instead of just `"`

```PO
#: src/window.ui:331
msgid ""
"The default model to use on new chats and when Alpaca is launched with the "
"option --ask \"message\""
msgstr ""
"El modelo por defecto a usar en chats nuevos y cuando Alpaca es lanzado con "
"la opción --ask \"mensaje\""
```

----

You can use a translator or AI to translate the model descriptions and update notes but please make sure the interface and flathub metadata are good, you can take all the time you need to translate it, I don't set version or time limits for translations, so don't worry about that.

Once you've translated everything please open a PR and I'll check the translation for anything out of place, once everything is checked and pushed into the project I will add your name to the credits in the app itself and the repo's README, if you want me to use a different name than the one you have on GitHub feel free to tell me. Thanks for your help, if you have any questions let me know in this thread!

**Reply by sh1d0re** [reply]
Hi, I'm currently around 15% done with the translation. It was difficult to simultaneously do the translation since I have another internship program currently progressing. I am currently in wanting of advice from you, just to check if 15% in 2 weeks is slow or not. I also just wanted to tell you I'm still working on it, and didn't abandon it lol. Thank you!

**Reply by Jeffser** [reply]
Don't worry about it you can take all the time you need!

**Reply by sh1d0re** [reply]
Happy new years! I'm done with the pull request. Tell me if there are any places to fix. Thank you!

**Comment by olumolu** [comment]
Provide translation update available asap so that can be made for 3.0

**Reply by Jeffser** [reply]
Just updated it

**Reply by olumolu** [reply]
Sure i will update them so it can be ready for next major

**Reply by Jeffser** [reply]
wait wait, I forgot about the patch notes, don't update your translation yet

**Reply by olumolu** [reply]
Sure Take your time

**Reply by Jeffser** [reply]
alright now they are ready

**Reply by UnconnectedBedna** [reply]
I'm around 50% done for swedish, and there is also things I need clarification on before implemented. I figured I do the complete translation and then solve the lines I have questions for in the PR..

Would you prefer a 50% translation to include it in the release?

**Edit**
NVM, I now realize the question was the other way around. LOL
Disregard this.
I'll keep working on it when I have time and provide the translation when done. xD

**Reply by olumolu** [reply]
You can push a pr with 50% translation done but you have to work on it as it need to be completed.

**Reply by UnconnectedBedna** [reply]
Ofc I can, but what would be the purpose for that?
I misinterpreted the initial request as being from Jeffser to immediately provide translations for next release. :smiley_cat:

**Reply by Henkster72** [reply]
I'm willing to translate to Dutch 🇳🇱

**Comment by Henkster72** [comment]
I'm willing to translate to Dutch 🇳🇱

**Reply by Jeffser** [reply]
I'll make a file for you don't worry

**Reply by Jeffser** [reply]
Ok [this](https://github.com/Jeffser/Alpaca/blob/main/po/nl.po) is the file

I recommend you use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to translate the strings, you can also do it manually, it's a plain text file.

Once you are finished open a pull request with the file, I'll take a look at it and then push it into the project, of course I will credit you in the app and repository!

If you have any questions feel free to ask.

**Reply by Henkster72** [reply]
Done. See the pull request

**Reply by olumolu** [reply]
I couldn't see any pr

**Reply by Henkster72** [reply]
Strange.. But now you can @olumolu ;)

**Comment by Henkster72** [comment]
@Jeffser 2 things: 1) Does the language of Alpaca always follow the system language? How about someone can change their language into one he prefers (in Preferences)? 2) Correct me if I am wrong, but the translations for every new models description is automated or should the language translator update every new description model?

**Reply by Jeffser** [reply]
Hi, the translations follow the system language and the translations for the models descriptions are added manually, but if you prefer me to autotranslate the model descriptions for Dutch I could do it.

**Reply by Henkster72** [reply]
Thank you for clarifying! I was just wondering if I had missed an option in the Alpaca GUI. My system language is set to English, so switching it to Dutch just to check the GUI translations wouldn't be ideal.

Regarding automated translations, I believe it’s definitely feasible since English generally translates well to Dutch. That said, I’d recommend avoiding tools like Google Translate for this purpose, as they often miss subtleties and nuances in translation. Keep up the great work—it’s appreciated!

**Comment by sokundo** [comment]
i can do arabic

**Reply by Jeffser** [reply]
Hi I just made [this file](https://github.com/Jeffser/Alpaca/commit/1e5dd9f7e87d15738c86c27a61931c045c0cab79) for the arabic translation.

You can use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to translate the strings

You can use Google translator or an AI to translate model descriptions and release notes, just make sure they make sense.

If you have any questions feel free to ask!

**Reply by sokundo** [reply]
alright i'm working on it, sorry in advance if i take too long as i don't have alot of free time to dedicate, though i will try to have it ready within a month or so.

also i have a question, some translations may need to be rephrased depending on the context they are show in to fit better, would it be fine to push a first draft and edit a few thing later if needed?(ofc it will be functional from the get go, this is just for polish) or should i hold off until i make sure everything is as good as i think it can be?

**Comment by harimanish** [comment]
@Jeffser I would like to add tamil  translations

**Reply by Jeffser** [reply]
Hi thanks for your interest, I made [this file](https://github.com/Jeffser/Alpaca/blob/main/po/ta.po) for the Tamil translations

I recommend you use GTranslator to work on the file, you can use AI for the release notes and model descriptions since there are a lot of strings that come from there.

Once it's done please do a PR and I'll check it out, if everything seems fine I'll turn it on and add your name to the app credits and the readme.

If you have any questions feel free to ask.

**Reply by harimanish** [reply]
sorry  @Jeffser for being late but this was a huge file

**Reply by olumolu** [reply]
There are some additional lines added update the repo

**Reply by harimanish** [reply]
Will do it

**Comment by nofalbriansah** [comment]
I want translate it to Indonesian

**Reply by Jeffser** [reply]
Hi, thanks for your interest, I made [this file](https://github.com/Jeffser/Alpaca/blob/main/po/id.po) for the Indonesian translations

I recommend using [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to work on this file, you can use AI or a translator to do the release notes and model descriptions since a there are a lot of strings from there.

Once you finish please do a PR and I'll check it out, if everything seems fine I'll turn on the translation and add your name to the credits in the app and the readme!

If you have any quesstions feel free to ask.

**Comment by Girish4489** [comment]


**Reply by mags0ft** [reply]
The ISO language code for Kannada is `kn`. `ka` is already in the process of being translated and stands for the Georgian language, Kartvelian.

**Comment by EkaterinePapava** [comment]
Hello,
I would like to do Georgian translation.
ISO language code for Georgian is 'ka'

Ekaterine

**Comment by 8ar10der** [comment]
Hi @Jeffser 

I found you using "Actions" in instead of "Tools" to type the models form the ollama. Could you explain more what you are trying to represent to the user?

Refer to the code line:
https://github.com/Jeffser/Alpaca/blob/9ec98668eddf04a014ace448ae87f9a1161e4e13/src/custom_widgets/model_manager_widget.py#L474C11-L474C88

I am asking because "actions" map to some different definition in Chinese

**Reply by Jeffser** [reply]
Honestly I just thought Actions sounded more indicative of the objective of the feature. Do you think I should change it back to tools?

**Reply by Jeffser** [reply]
Forget it, it was a dumb change anyways, I'm renaming everything to `tools`

**Comment by martinkolenic** [comment]
Hi there, I'd like to provide a translation to Slovak (sk). Great job with the app, by the way, I really like it :)

**Reply by olumolu** [reply]
Download this file https://github.com/Jeffser/Alpaca/blob/main/po/alpaca.pot
Rename the file as sk.po and put your translation into msgstr " " and keep msgid same.
We recommend using GTranslator to work on this file, you can use AI or a translator to do the release notes and model descriptions since a there are a lot of strings from there.

Once you finish please do a PR.

If you have any quesstions feel free to ask.

**Comment by roche-jeethan** [comment]
I can get started with Kannada, which is an Indian Language
From Karnataka, India

**Reply by roche-jeethan** [reply]
Translation 20% complete. Estimated for complete translation to Kannada in 3 days

**Reply by roche-jeethan** [reply]
Translation 33.33% complete. Estimated for completion to Kannada in 2 days

**Reply by roche-jeethan** [reply]
Translation 50% complete. Estimated for completion to Kannada in 1.5 days

**Reply by roche-jeethan** [reply]
Translation 100% complete. Pull request has been initialized without conflicts.

**Comment by vishnuu-1104** [comment]
Hello Jeffser, I would like to do translation in Malayalam (India)

**Reply by olumolu** [reply]
> Hello Jeffser, I would like to do translation in Malayalam (India)

you can fork the project and make a file called ml.po and copy paste the whole contain of alpaca.pot from po dir and start your translation into msgstr and once done you can just create a merge request after verify it will be merged.

**Comment by PlagaMedicum** [comment]
Hi! I would like to add Belarusian language!

**Reply by PlagaMedicum** [reply]
the code is just "be". Will you create a file or I can handle it myself?

**Reply by olumolu** [reply]
just take https://github.com/Jeffser/Alpaca/blob/main/po/alpaca.pot and do your translation and after name make a pull request

**Comment by giannisjv** [comment]
hey, I would love to do a Greek translation....

**Reply by Jeffser** [reply]
Hi thanks for posting this here after your email yesterday!

I prepared the file [here](https://github.com/Jeffser/Alpaca/blob/main/po/el.po)

You can use [GTranslator](https://flathub.org/apps/org.gnome.Gtranslator) to manipulate the file or just edit it with a text editor.

You can take however long you need to fill the lines. You don't have to translate the patch notes if you don't want to btw.

Thanks again!

**Comment by Henkster72** [comment]
I see that the Dutch translation is gone? Why? Is there a major update on nl.po? How can I look for what to add/change with an update of Alpaca?

**Reply by Jeffser** [reply]
MY BAAAAD.

I made a thingy that deleted inactive translations and it deleted dutch by mistake.

I'll reupload the last version!

**Reply by Jeffser** [reply]
PS: By inactive I mean translations that didn't have any actual translations, so empty files

**Reply by Jeffser** [reply]
Wait no, the file is there, what do you mean then?

**Reply by Henkster72** [reply]
Where can I find the translation? 
In the app I can not find the dutch translation?
Also I believe a lot is changed since I translated to Dutch (about 750 lines)? Can I get a new nl.po to translate? Or how does that work? Or maybe can anyone get an "empty" po file with every update to fill in?

**Reply by Jeffser** [reply]
I'll update the files, the dutch translation should be turned on for people with dutch set as their device language

**Comment by T0F1Q2007** [comment]
I want to contribute for Azerbaijani language

**Comment by lloura** [comment]
Hi! I’d like to help the project by updating the Brazilian Portuguese translation file. I’ve already forked the project and ran the translation_tool.sh script (though I’m not entirely sure what it did, other than updating a lot of seemingly random lines xD).
Before I start actually updating the strings and eventually making a pull request, I wanted to ask if there’s anything else I should do or keep in mind.
Thanks!

**Reply by Jeffser** [reply]
Hi thanks for the help, the brazilian portuguese file is `po/pt_BR.po`, you can update as many lines as you want to, there's this tool called [GTranslator](https://flathub.org/en/apps/org.gnome.Gtranslator) that you can use to write the empty lines, if you have any questions feel free to ask
